### PR TITLE
Nj 154268851 storage pods

### PR DESCRIPTION
**Pull request by nehagjain15**
Allow DatastoreClusters to be specified in vm_type and use that to create vm's and ephemeral disks. 
DO NOT MERGE. WIP

### CODE REVIEW

File: src/vsphere_cpi/spec/integration/cluster_cloud_properties_spec.rb [hunk 1]
```diff
@@ -165,4 +165,93 @@
...
+            ]
+          }
+        ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    let(:disk_pool) { { 'datastores' => [@disjoint_datastore] } }
+
+    it 'should place disk into datastore that belongs to the cluster defined in cloud config' do
```
Comment (cppforlife) [inline src/vsphere_cpi/spec/integration/cluster_cloud_properties_spec.rb]: we should probably asset that stemcell is also copied into particular datastore as well

File: src/vsphere_cpi/lib/cloud/vsphere/cloud.rb [hunk 2]
```diff
@@ -506,15 +506,38 @@ def delete_disk(raw_director_disk_cid)
...
+          original_stemcell_vm,
+          name_of_replicated_stemcell,
+          @datacenter.template_folder.mob,
+          cluster.resource_pool.mob,
+          datastore_cluster: datastore_cluster
+        )
+        @logger.info("Recommendation from Storage DRS for: #{name_of_replicated_stemcell}, Destination: #{recommendation.action.first.destination.name}")
+        #loop over all actions to pick one with reason as storagePlacement?
+        if recommendation.reason == 'storagePlacement' && recommendation.action.first.destination.class == VimSdk::Vim::Datastore
+          to_datastore = recommendation.action.first.destination
```
Comment (cppforlife) [inline src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: could there be other recommendations (not first)?

Comment (nehagjain15) [reply src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: RecommendDatastores can return multiple recommendations, the best one will be first entry in the array.

File: .gitignore [hunk 3]
```diff
@@ -11,6 +11,7 @@ blobs
 .DS_Store
 .final_builds/jobs/**/*.tgz
 .final_builds/packages/**/*.tgz
+.byebug_history
```
Comment (ktchen14) [inline .gitignore]: This should go in your global gitignore rather than in this project specific one.

**Pull request merged**